### PR TITLE
Issue/4609

### PR DESCRIPTION
**Pull request by ursa-h**
[Allow Set Library Search Order to override the local keywords precedence.](https://github.com/robotframework/robotframework/issues/4609#top)

Restore ability to better control the prioritized keywords through Library Search Order
- Prioritize search order even when there is another keyword that is
  - a better match 
  - local 
  - private
 
Enables simpler unit testing of keywords

### CODE REVIEW

**Comment by pekkaklarck** [comment]
Is there some reason for this PR to be Draft? On a quick look it looks pretty good. Very nice work updating tests. Few comments about it anyway:

- New tests should be added for a situation where there are multiple matches after applying the search order. That can happen at least if there are multiple keywords having embedded arguments that match the used keyword in the preferred resource/library. Tests should be organized so that one of the keywords matches better than others. If there already are tests like this and I missed them, let me know.
- New tests should be added for a situation where there's an unresolvable conflict after applying the search order.
- Do you remember is there documentation somewhat that should be updated?
- Some of the edited files don't have newline at the end. Please add them to make the diff (and future diffs if there are new edits) cleaner.

RF 6.1 is already pretty massively late and we try to get it released ASAP. It would be nice to get this into that release, because otherwise it needs to wait until RF 6.2 or RF 7.0, whichever comes first. I hope we'd be close to the release candidate after the next week. Ping me on Slack if you have problems hitting that deadline.

**Comment by ursa-h** [comment]
Thank you for the valuable comments @pekkaklarck 

4 tests added:
Best Search Order Controlled Match Wins In Resource
Best Search Order Controlled Match Wins In Library
Search order causes conflict within resource
Search order causes conflict within library

In documentation I changed only one sentence. I wonder if I missed something. It all looked like it now works.

File: doc/userguide/src/CreatingTestData/CreatingUserKeywords.rst [hunk 1]
```diff
@@ -702,9 +702,9 @@ succeeds:
 
 Before looking which match is best, Robot Framework checks are some of the matching
 keywords implemented in the same file as the caller keyword. If there are such keywords,
-they are given precedence over other keywords. If there are still conflicts
-after looking for best matches, Robot Framework checks can they be
-resolved based on the `library search order`_.
+they are given precedence over other keywords. Alternatively, `library search order`_
```
Comment (ursa-h) [inline doc/userguide/src/CreatingTestData/CreatingUserKeywords.rst:705]: Link does not seem to work. Could be replaced with `:name:`Set Library Search Order` from the BuiltIn_ library`

**Comment by pekkaklarck** [comment]
Looks great! I enabled running tests and I'll merge this if they pass.

There are some whitespace issues with tests (missing newlines, tabs instead of spaces), but I can fix them myself. I'll also check the issue with link.

**Comment by pekkaklarck** [comment]
The link in documentation worked fine. Notice that you need to generate the User Guide using the `doc/userguide/ug2html.py` script to see how changes actually look like. I already fixed whitespace issues locally.

**Pull request merged**